### PR TITLE
Adding changes for rbac for cert rotation

### PR DESCRIPTION
**Pull request by sowmyav27**
## Issue: 
Adding automated tests for : https://github.com/rancher/qa-tasks/issues/409
 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: tests/v2/validation/rbac/rbac_cert_rotation_test.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+			require.NoError(rb.T(), err)
+			rb.standardUser = newUser
+			rb.T().Logf("Created user: %v", rb.standardUser.Username)
+			rb.standardUserClient, err = rb.client.AsUser(newUser)
+			require.NoError(rb.T(), err)
+
+			subSession := rb.session.NewSession()
+			defer subSession.Cleanup()
+
+			createProjectAsAdmin, err := createProject(rb.client, rb.cluster.ID)
```
Comment (anupama2501) [inline tests/v2/validation/rbac/rbac_cert_rotation_test.go:117]: Just a thought. Can we remove project creation from the for loop and move it above for loop? so we are not creating a new project for Cluster Owner, cluster member and restricted admin as they are not utilizing the created project. By moving this above for loop, we are making sure to create a single project and make use of it for ProjectOwner and ProjectMember.

Or if we want a new project for PO and PM we can move this block to line 126..

File: tests/v2/validation/rbac/rbac_cert_rotation_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+		rb.Run("Test case - RotateCertificates of a cluster as a "+tt.name, func() {
+			rb.ValidateCertRotationCluster(tt.role)
+		})
+
+	}
+}
+
+func TestCertRotationRbacTestSuite(t *testing.T) {
+	logrus.Infof("Beginning testing")
```
Comment (anupama2501) [inline tests/v2/validation/rbac/rbac_cert_rotation_test.go:149]: can we remove the log here? as we have "Starting validations" log added in the TestCertRotationRbac?

Comment (igomez06) [reply tests/v2/validation/rbac/rbac_cert_rotation_test.go:149]: agreed

File: tests/v2/validation/rbac/rbac_cert_rotation_test.go (hunk 1 above)
Comment (igomez06) [reply tests/v2/validation/rbac/rbac_cert_rotation_test.go:117]: agreed

File: tests/framework/extensions/certs/rotatecerts/rotatecerts.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+)
+
+// RotateCertificates is a helper function to rotate the certificates on an RKE2 or k3s cluster. Returns error if any.
+func RotateCertificates(client *rancher.Client, clustername string, namespace string) error {
+
```
Comment (igomez06) [inline tests/framework/extensions/certs/rotatecerts/rotatecerts.go:13]: Looks great. Can we remove this extra return.

**Pull request closed without merging**